x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/SearchApplication.java

```diff
@@ -139,7 +138,7 @@ public void writeTo(StreamOutput out) throws IOException {
 
     static {
         PARSER.declareStringOrNull(optionalConstructorArg(), NAME_FIELD);
-        PARSER.declareStringArray(constructorArg(), INDICES_FIELD);
+        PARSER.declareStringArray(optionalConstructorArg(), INDICES_FIELD);
```

Comment:
Do we want to check on the transport version here, and require the indices field/have the same serialization for older versions of the transport serialization? That would probably be the safest way to go, especially since we should bump the transport version anyway with this serialization change.

Reply:
It's not clear to me whether it's required to check for the transport version because the feature is still in beta and subject to change.

Reply:
We don't need to change serialization as the indices field has not been removed. IMO, I think we need to remove it and change serialization.

Reply:
I agree with @carlosdelest - keeping the indices field given our decisions seems like it's introducing tech debt.